### PR TITLE
EPMKHRDMH-346 User can delete predefined templates (sick, vac) without knowing this

### PR DESCRIPTION
**Pull request by lotrien**
- removed ability to delete predefined events (sick, vac);
- add handle for case: "No Tasks Templates shows" when no tasks templates but event templates exist;
- small refactor;

THIS PR SHOULD BE MERGED AFTER ALL PREVIOUS;

### CODE REVIEW

File: public/modules/templates/services/templates.task.client.service.es6 [hunk 1]
```diff
@@ -1,11 +1,14 @@
 class TemplatesService {
-	getById(user, id, type) {
-		return user[type].find(template => template._id == id);
-	}
+    getById(user, id, type) {
+        return user[type].find(template => template._id == id);
```
Comment (make-it-happen-team) [inline public/modules/templates/services/templates.task.client.service.es6]: Use strict equals

File: public/modules/templates/controllers/templates.task.client.controller.es6 [hunk 2]
```diff
@@ -16,18 +16,19 @@ class TemplatesController {
...
 
 		updatedUser.$update(updatedUser => {
-			this.Notification.success(`"Template ${template.title}" was successfully updated`);
+			this.Notification.success(`'Template ${template.title}' was successfully updated`);
 			this.user = this.Authentication.user = updatedUser;
 		}, err => console.error(err));
 	}
 
 	remove(templateToRemove, templateType) {
+		debugger;
```
Comment (make-it-happen-team) [inline public/modules/templates/controllers/templates.task.client.controller.es6]: Why this is not remove?

**Review by make-it-happen-team — CHANGES_REQUESTED** [review]

File: public/modules/templates/controllers/templates.task.client.controller.es6 (hunk 2 above)
Comment (lotrien) [reply public/modules/templates/controllers/templates.task.client.controller.es6]: this was removed in latest commits. so currently it is not in this file

File: public/modules/templates/services/templates.task.client.service.es6 (hunk 1 above)
Comment (lotrien) [reply public/modules/templates/services/templates.task.client.service.es6]: fixed

**Pull request merged**